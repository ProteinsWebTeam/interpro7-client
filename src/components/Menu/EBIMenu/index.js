// @flow
import React from 'react';
import T from 'prop-types';

import MenuItem from 'components/Menu/MenuItem';

import { EBI } from 'menuConfig';

import { foundationPartial } from 'styles/foundation';
import fonts from 'EBI-Icon-fonts/fonts.css';

const iconStyle = foundationPartial(fonts);

const EBIMenu = (
  { className, children } /*: {className?: string, children?: any} */,
) =>
  <ul className={className}>
    {children}
    {EBI.map(({ href, name, icon, iconClass = 'generic' }) =>
      <li key={href}>
        <MenuItem href={href}>
          {icon &&
            <i
              data-icon={icon}
              className={iconStyle('icon', `icon-${iconClass}`)}
            />}
          {name}
        </MenuItem>
      </li>,
    )}
  </ul>;
EBIMenu.propTypes = {
  className: T.string,
  children: T.any,
};

export default EBIMenu;
