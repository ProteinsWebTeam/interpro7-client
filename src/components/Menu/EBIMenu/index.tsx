import React, { PropsWithChildren } from 'react';

import MenuItem from 'components/Menu/MenuItem';

import { EBI } from 'menuConfig';

import cssBinder from 'styles/cssBinder';
import fonts from 'EBI-Icon-fonts/fonts.css';

const iconStyle = cssBinder(fonts);

const EBIMenu = ({
  className,
  children,
}: PropsWithChildren<{ className?: string }>) => (
  <ul className={className}>
    {children}
    {EBI.map(({ href, name, icon, iconClass = 'generic' }) => (
      <li key={href}>
        <MenuItem href={href}>
          {icon && (
            <i
              data-icon={icon}
              className={iconStyle('icon', `icon-${iconClass}`)}
            />
          )}
          {name}
        </MenuItem>
      </li>
    ))}
  </ul>
);

export default EBIMenu;
