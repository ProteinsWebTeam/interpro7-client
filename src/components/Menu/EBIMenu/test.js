import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import EBIMenu from '.';

const renderer = new ShallowRenderer();

describe('<EBIMenu />', () => {
  test('should render', () => {
    renderer.render(<EBIMenu />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
