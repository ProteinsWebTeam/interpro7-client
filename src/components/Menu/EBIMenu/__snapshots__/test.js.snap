// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EBIMenu /> should render 1`] = `
<ul>
  <li>
    <Memo(ConnectFunction)
      href="https://www.ebi.ac.uk"
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      EMBL-EBI
    </Memo(ConnectFunction)>
  </li>
  <li>
    <Memo(ConnectFunction)
      href="https://www.ebi.ac.uk/services"
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      Services
    </Memo(ConnectFunction)>
  </li>
  <li>
    <Memo(ConnectFunction)
      href="https://www.ebi.ac.uk/research"
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      Research
    </Memo(ConnectFunction)>
  </li>
  <li>
    <Memo(ConnectFunction)
      href="https://www.ebi.ac.uk/training"
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      Training
    </Memo(ConnectFunction)>
  </li>
  <li>
    <Memo(ConnectFunction)
      href="https://www.ebi.ac.uk/about"
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      About EBI
    </Memo(ConnectFunction)>
  </li>
</ul>
`;
